sdk/eventgrid/azure-messaging-eventgrid/src/main/java/com/azure/messaging/eventgrid/EventGridEvent.java

```diff
@@ -127,7 +159,18 @@ public String getSubject() {
      * @return A {@link BinaryData} that wraps the this event's data payload.
      */
     public BinaryData getData() {
-        return EventGridDeserializer.getData(event.getData());
+        if (this.binaryData == null) {
+            Object data = this.event.getData();
+            if (data instanceof byte[]) {
+                this.binaryData = BinaryData.fromBytes((byte[]) data);
+            }
+            if (data instanceof String) {
```

Comment:
Could this go as `else if`  ?

Reply:
Another PR for EventGridEvent and other stuffs will be created soon. This part will be changed.
Let's discuss in there.